Add JRuby 10.0.0.0

headius opened the pull request:
Add JRuby 10 and set it as default! Hooray!

headius commented:
I'm not sure what those failing matches are supposed to do nor why they are failing. Help?

pkuczynski commented:
`[alternative-interpreters-tests (macos-13, jruby)` failes because `rvm jruby do ruby -v` returns a string matching `/jruby/` but not `/OpenJDK/`. Has this been removed from the output?

Can you then update the test please?

headius commented:
I don't know what would have changed. Locally, it still looks the same as earlier releases to me:

```
jruby 9.4.12.0 (3.1.4) 2025-02-11 f4ab75096a OpenJDK 64-Bit Server VM 21.0.6+7-LTS on 21.0.6+7-LTS +jit [arm64-darwin]
jruby 10.0.0.0 (3.4.2) 2025-05-07 3eaf87bd37 OpenJDK 64-Bit Server VM 21.0.6+7-LTS on 21.0.6+7-LTS +indy +jit [arm64-darwin]
```

headius commented:
It also matches locally using rvm for me:

```
[] rvm $ tmp/bin/rvm install jruby-10.0.0.0
Searching for binary rubies, this might take some time.
Found remote file https://repo1.maven.org/maven2/org/jruby/jruby-dist/10.0.0.0/jruby-dist-10.0.0.0-bin.tar.gz
Checking requirements for osx.
Installing requirements for osx.
Updating system.......
Installing required packages: make..
Requirements installation successful.
jruby-10.0.0.0 - #configure
jruby-10.0.0.0 - #download
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 32.1M  100 32.1M    0     0  17.8M      0  0:00:01  0:00:01 --:--:-- 17.8M
No checksum for downloaded archive, recording checksum in user configuration.
jruby-10.0.0.0 - #validate archive
jruby-10.0.0.0 - #extract
jruby-10.0.0.0 - #validate binary
jruby-10.0.0.0 - #setup
jruby-10.0.0.0 - #gemset created /Users/headius/work/rvm/tmp/gems/jruby-10.0.0.0@global
jruby-10.0.0.0 - #importing gemset /Users/headius/work/rvm/tmp/gemsets/jruby/global.gems..
jruby-10.0.0.0 - #generating global wrappers........
jruby-10.0.0.0 - #gemset created /Users/headius/work/rvm/tmp/gems/jruby-10.0.0.0
jruby-10.0.0.0 - #importing gemsetfile /Users/headius/work/rvm/tmp/gemsets/default.gems evaluated to empty gem list
jruby-10.0.0.0 - #generating default wrappers........
[] rvm $ tmp/bin/rvm jruby-10.0.0.0 do ruby -v
jruby 10.0.0.0 (3.4.2) 2025-04-13 6ed59bc847 OpenJDK 64-Bit Server VM 21.0.6+7-LTS on 21.0.6+7-LTS +indy +jit [arm64-darwin]
```

headius commented:
I have pushed a new PR that additionally adds 10.0.0.1 and 9.4.12.1 here: https://github.com/rvm/rvm/pull/5567

It is based on this branch, so will need to rebase if this one is updated for some CI fix.

I am not sure how to proceed on the CI issue.

headius commented:
Help? Should I try to push some experimental commits to the CI process to figure this out?

headius commented:
I think I figured it out: the default Java on GHA is still 17, but JRuby requires 21. Adding `setup-java` action solves that.

I made the change on the tip of #5567 branch, and it now seems like it will be [green](https://github.com/rvm/rvm/actions/runs/14894217135) (but it is stalled at the moment). Would you prefer if I moved that change here and rebased the 10.0.0.1 PR? Let me know how you want me to proceed.

pkuczynski commented:
> I think I figured it out: the default Java on GHA is still 17, but JRuby requires 21. Adding `setup-java` action solves that.
> 
> I made the change on the tip of #5567 branch, and it now seems like it will be [green](https://github.com/rvm/rvm/actions/runs/14894217135) (but it is stalled at the moment). Would you prefer if I moved that change here and rebased the 10.0.0.1 PR? Let me know how you want me to proceed.

I saw your changes, they look good! I think we should port the change to this PR, since it was the first.

In a separate PR we could/should add a check for the correct java version when latest jruby is installed:
* https://github.com/rvm/rvm/blob/master/scripts/functions/requirements/osx_brew#L381 > https://github.com/rvm/rvm/blob/master/scripts/functions/osx-support#L7
* https://github.com/rvm/rvm/blob/master/scripts/functions/requirements/debian#L179

headius commented:
I moved the change here and it's green, so I rebased #5567 on this change. Let me know if you need me to do anything else!

pkuczynski commented:
> I moved the change here and it's green, so I rebased #5567 on this change. Let me know if you need me to do anything else!

In this PR is all good so let's merge it. But maybe in a separate one we can add warnings in the requirements script when Java is too low?
